### ISSUE TITLE
Our fuzzers broke with the latest external API changes

### ISSUE THREAD

**Issue by isislovecruft** [isislovecruft/curve25519-dalek#91]
We changed enough of our external API in the last version (possibly earlier) that our fuzzers are broken. The fuzzer for `DecafPoint`s, now `RistrettoPoint`s, I believe cannot be usefully recovered since it expects to directly manipulate FieldElements in an unsound way which we've now made impossible to external applications by making them `pub(crate)`. (It's a little annoying that we can only fuzz the external API.)

For the case of `scalar_constructor_accepts_256bit_values`:

1. we don't accept 256-bit values anymore
2) we *do* accept values in (2²⁵²-27742317777372353535851937790883648493,2²⁵⁵-1] but it should be pretty difficult to construct one of these and use it mistakenly

So it doesn't seem all that useful because I don't see anyway this could not be the case, but we could change the scalar fuzzer to check that:

```rust
    // Compute c = a*b (mod l)
    let c1 = (&Scalar::from_bits(a_bytes) * &Scalar::from_bits(b_bytes)).reduce();

    // Compute c = (a mod l) * (b mod l)
    let a_mod_l = Scalar::from_bytes_mod_order(a_bytes);
    let b_mod_l = Scalar::from_bytes_mod_order(b_bytes);
    let c2 = &a_mod_l * &b_mod_l;

    assert_eq!(c1, c2);                                                                          
```

**Comment by hdevalence** [comment]
I'm not sure how much benefit we actually get from fuzzers versus just designing a safe API, as you mention.  I think we should just remove the fuzzers we have.

**Comment by isislovecruft** [comment]
Fixed in #93
